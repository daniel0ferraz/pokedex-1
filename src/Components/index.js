import Image from "./Image";
import Search from "./Search";
import PokemonCard from "./PokemonCard";
import Loading from "./Loading";
import PokemonInfo from "./PokemonInfo";
import Type from "./Type";

export { Image, Search, PokemonCard, Loading, PokemonInfo, Type };
