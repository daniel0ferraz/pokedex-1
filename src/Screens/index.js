import Home from "./Home";
import Profile from "./Profie";

export { Home, Profile };
