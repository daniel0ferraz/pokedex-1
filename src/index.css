@font-face {
  font-family: "SF Pro Display";
  src: local("SF Pro Display"),
    url(./assets/fonts/SFProDisplay-Regular.ttf) format("truetype");
}

* {
  background-color: transparent;
  box-sizing: border-box;
  border: none;
  margin: 0;
  padding: 0;
  outline: none;
}
body {
  font-family: SF Pro Display;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

button {
  cursor: pointer;
}
