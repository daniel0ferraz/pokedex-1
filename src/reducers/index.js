import { combineReducers } from "redux";
import { pokedexReducer } from "./pokedex";

export const Reducers = combineReducers({ pokedex: pokedexReducer });
