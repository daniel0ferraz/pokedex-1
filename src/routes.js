import React, { useEffect } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Home, About } from "./Screens";
import { connect } from "react-redux";
import { gottaCatchThemAll } from "./actions/pokedex";

function Routes(props) {
  const { limit, offset, pokemons, gottaCatchThemAll } = props;

  useEffect(() => {
    if (!pokemons) gottaCatchThemAll(limit, offset);
  }, [gottaCatchThemAll, limit, offset, pokemons]);
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/about/:id" component={About} />
      </Switch>
    </Router>
  );
}

const mapStateToProps = (state) => ({
  limit: state.pokedex.limit,
  offset: state.pokedex.offset,
  pokemons: state.pokedex.pokemons,
});

const mapDispatchToProps = (dispatch) => ({
  gottaCatchThemAll: (limit, offset) =>
    dispatch(gottaCatchThemAll(limit, offset)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Routes);
