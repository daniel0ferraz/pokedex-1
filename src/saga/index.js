import { all } from "redux-saga/effects";
import { watchPokedex } from "./pokedex";

export default function* Saga() {
  yield all([watchPokedex()]);
}
