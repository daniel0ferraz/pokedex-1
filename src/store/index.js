import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { Reducers } from "../reducers";
import Saga from "../saga";
import { apiMiddleware } from "redux-api-middleware";
import { composeWithDevTools } from "redux-devtools-extension";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  Reducers,
  composeWithDevTools(applyMiddleware(sagaMiddleware, apiMiddleware))
);

sagaMiddleware.run(Saga);

export { store };
