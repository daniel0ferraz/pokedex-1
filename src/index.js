import React from "react";
import ReactDOM from "react-dom";
import "./index.css";
import { Provider } from "react-redux";
import { store } from "./store";
import * as serviceWorker from "./serviceWorker";
import Routes from "./routes";

ReactDOM.render(
  <React.StrictMode>
    <Provider store={store}>
      <Routes />
    </Provider>
  </React.StrictMode>,
  document.getElementById("root")
);

serviceWorker.unregister();
