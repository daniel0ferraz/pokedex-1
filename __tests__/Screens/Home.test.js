// Link.react.test.js
import React from "react";
import "babel-polyfill";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../src/store";
import { Home } from "../../src/Screens";

test("Rendering Home", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
