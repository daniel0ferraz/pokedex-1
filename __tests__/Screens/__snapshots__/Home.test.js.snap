// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Home 1`] = `
Array [
  <div
    className="sc-fznyAO vZIGY"
  >
    <div
      className="sc-AxjAm dFYpwF sc-fznKkj fvgEGZ"
      size="100%"
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "width": "100%",
            "zIndex": 2,
          }
        }
      />
    </div>
  </div>,
  <div
    className="sc-fzokOt fCyYDE"
  >
    <div
      className="sc-fzoiQi dAGsKk"
    />
  </div>,
]
`;
